lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -0,0 +1,408 @@
+        default: present
+        choices: [ present, absent, restarted, started, stopped ]
+        type: str
+extends_documentation_fragment: hcloud
+"""
+
+EXAMPLES = """
+# Create a basic server
+- name: Create a basic server
+  hcloud_server:
+    name: my-server
+    server_type: cx11
+    image: ubuntu-18.04
+    state: present
+
+# Create a server in an specific location and with ssh keys
+- name: Create a basic server with ssh key
+  hcloud_server:
+    name: my-server
+    server_type: cx11
+    image: ubuntu-18.04
+    location: fsn1
+    ssh-key: my-ssh-key
+    state: present
+
+# Resize an existing server
+- name: Resize an existing server
+  hcloud_server:
+    name: my-server
+    server_type: cx21
+    keep_disk: true
+    state: present
+
+# Remove a server
+- name: server should be absent
+  hcloud_server:
+    name: my-server
+    state: absent
+
+# Start a server
+- name: server should be started
+  hcloud_server:
+    name: my-server
+    state: started
+
+# Stop a server
+- name: server should be stopped
+  hcloud_server:
+    name: my-server
+    state: stopped
+
+# Restart a server
+- name: server should be restarted
+  hcloud_server:
+    name: my-server
+    state: restarted
+"""
+
+RETURN = """
+hcloud_server:
+    description: The server instance
+    returned: Always
+    type: dict
+    sample: {
+        "backup_window": null,
+        "datacenter": "nbg1-dc3",
+        "id": 1937415,
+        "image": "ubuntu-18.04",
+        "ipv4_address": "116.203.104.109",
+        "ipv6": "2a01:4f8:1c1c:c140::/64",
+        "labels": {},
+        "location": "nbg1",
+        "name": "mein-server-2",
+        "rescue_enabled": false,
+        "server_type": "cx11",
+        "status": "running"
+    }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.hcloud import Hcloud
+
+try:
+    from hcloud.volumes.domain import Volume
+    from hcloud.ssh_keys.domain import SSHKey
+    from hcloud.servers.domain import Server
+    from hcloud import APIException
+except ImportError:
+    pass
+
+
+class AnsibleHcloudServer(Hcloud):
+    def __init__(self, module):
+        Hcloud.__init__(self, module, "hcloud_server")
+        self.hcloud_server = None
+
+    def _prepare_result(self):
+        return {
+            "id": to_native(self.hcloud_server.id),
+            "name": to_native(self.hcloud_server.name),
+            "ipv4_address": to_native(self.hcloud_server.public_net.ipv4.ip),
+            "ipv6": to_native(self.hcloud_server.public_net.ipv6.ip),
+            "image": to_native(self.hcloud_server.image.name),
+            "server_type": to_native(self.hcloud_server.server_type.name),
+            "datacenter": to_native(self.hcloud_server.datacenter.name),
+            "location": to_native(self.hcloud_server.datacenter.location.name),
+            "rescue_enabled": to_native(self.hcloud_server.rescue_enabled),
```

Comment:
This would turn everything into a string, not sure if that's what you want for e.g. booleans).
Same for the other values that are not strings. Probably not what you want.